packages/api/src/user.js

```diff
@@ -569,6 +569,43 @@ export async function userPinsGet (request, env) {
   }, { headers })
 }
 
+/**
+ *  List a user's pins regardless of the token used.
+ *  As we don't want to scope the Pinning Service API to users
+ *  we need a new endpoint as an umbrella.
+ *
+ * @param {import('./user').AuthenticatedRequest} request
+ * @param {import('./env').Env} env
+ * @param {import('./index').Ctx} ctx
+ * @return {Promise<JSONResponse>}
+ */
+export async function userPinDelete (request, env, ctx) {
+  // @ts-ignore
+  const requestId = request.params.requestId
+
+  if (!requestId) {
+    throw new PSAErrorRequiredData(REQUIRED_REQUEST_ID)
+  }
+
+  if (typeof requestId !== 'string') {
+    throw new PSAErrorInvalidData(INVALID_REQUEST_ID)
+  }
+
+  // TODO: Improve me, it is un-optimal to get the tokens in a separate request to the db.
+  // @ts-ignore
+  const tokens = (await env.db.listKeys(request.auth.user._id, { includeDeleted: true })).map((key) => key._id)
```

Comment:
The problem here is the function annotation rather than the way we pass the value. (hence you see `@ts-ignore` everywhere)

In fact, we should always use `string` rather than `number` when it comes to referencing values mapped to `bigint` in our DB. 
The reason is `bigint` columns in PG can hold values that are bigger than JS `Number.MAX_SAFE_INTEGER`.

For that reason, `parseInt` might return the wrong value for big numbers.

ie. 
```
parseInt('9007199254740993') 
9007199254740992
```

I've updated the function annotation to avoid any confusion in the future!

Reply:
Nice! Yeah I could be handy to follow up, maybe in a separate ticket to ensure user_ids are all strings. I noticed a few times their throw around as Numbers 😨